libtock/lsm303dlhc.c

```diff
@@ -20,11 +20,22 @@ static void command_callback_yield (int data1, int data2, int data3, void* ud) {
 
 
 static int lsm303dlhc_subscribe (subscribe_cb cb, void *userdata) {
-  return subscribe(DRIVER_NUM_LSM303DLHC, 0, cb, userdata);
+  subscribe_return_t subval = subscribe2(DRIVER_NUM_LSM303DLHC, 0, cb, userdata);
+  if (subval.success == 0) {
+    return tock_error_to_rcode(subval.error);
+  }
+  return TOCK_SUCCESS;
 }
 
-static int lsm303dlhc_command (uint32_t command_num, uint32_t data1, uint32_t data2) {
-  return command(DRIVER_NUM_LSM303DLHC, command_num, data1, data2);
+static int lsm303dlhc_command_noval (uint32_t command_num, uint32_t data1, uint32_t data2) {
+  syscall_return_t com = command2(DRIVER_NUM_LSM303DLHC, command_num, data1, data2);
+  if (com.type == TOCK_SYSCALL_SUCCESS) {
+    return TOCK_SUCCESS;
+  } else if (com.type == TOCK_SYSCALL_FAILURE) {
+    return tock_error_to_rcode(com.data[0]);
+  } else {
+    return TOCK_EBADRVAL;
```

Comment:
Uh, does this mean I messed up all of my PRs? I copied from console, I thought that would be a good example to go by, and return `TOCK_FAIL`.

Reply:
Unfortunately I think `TOCK_EBADRVAL` was added partway through the porting process, so the vast majority of transitioned drivers do this the wrong way. Some even directly call `exit()`, I noticed. This would be good to address, but probably does not need to be blocking

Reply:
Yeah, let's fix this between alpha and RC1.